### PR TITLE
Issue#1035 - Assert Canonical Tag Link

### PR DESCRIPTION
**Pull request by juanmrad**
For #1035

### Issue:
For Marketing purposes we need to be able to assert canonical tag location.

### Solution:
Add function to assert canonical tag exist and contains proper link.

### CODE REVIEW

**Comment by juanmrad** [comment]
!force code review

**Comment by aserv92** [comment]
!force code review

**Comment by healthlabs-bugman** [comment]
Paging @fortoulrodolfo to review the front-end work in this pull request!
Paging @ramsey-medology to review the back-end work in this pull request!
Paging @Chekote to review the testing work in this pull request!

File: src/Behat/FlexibleMink/Context/FlexibleContext.php [hunk 1]
```diff
@@ -1213,6 +1214,25 @@ public function elementHasAttributeValues(NodeElement $node, array $attributes)
...
+        $nodes = $this->getSession()->getPage()->findAll($selector, $locator);
+
+        if (!count($nodes)) {
+            throw new ExpectationException("No node elements were found on the page using '$locator'", $this->getSession());
+        } elseif ($occurrences && count($nodes) != $occurrences) {
+            throw new ExpectationException("Expected $occurrences nodes with '$locator' but found " . count($nodes), $this->getSession());
+        }
+
+        foreach ($nodes as $node) {
+            $this->elementHasAttributeValues($node, $attributes);
```
Comment (Chekote) [inline src/Behat/FlexibleMink/Context/FlexibleContext.php]: This is not an assertion. The method is returning a boolean. You need to check the response and throw an exception if you want this to assert.

File: features/assertLinkLocation.feature [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@clearAlertsWhenFinished
+Feature: Alert context
+  In order to ensure that JavaScript alerts work as expected
```
Comment (fortoulrodolfo) [inline features/assertLinkLocation.feature]: Are we checking alerts in this feature file?? 🤔

File: src/Behat/FlexibleMink/PseudoInterface/LinkContextInterface.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Pseudo interface for tracking the methods of the ContainerContext.
+ */
+trait LinkContextInterface
+{
+    /**
+     * Asserts that the canonical tag points to the given location.
+     *
+     * @param  string               $destination The location the link should be pointint to.
+     * @throws ExpectationException
```
Comment (fortoulrodolfo) [inline src/Behat/FlexibleMink/PseudoInterface/LinkContextInterface.php]: Please add a description when the exception may occur.

File: src/Behat/FlexibleMink/Context/FlexibleContext.php (hunk 1 above)
Comment (ramsey-medology) [reply src/Behat/FlexibleMink/Context/FlexibleContext.php]: Or wrap in an `assertTrue` and provide a specific failure message

File: src/Behat/FlexibleMink/Context/LinkContext.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Behat\FlexibleMink\Context;
+
+use Behat\FlexibleMink\PseudoInterface\FlexibleContextInterface;
+use Behat\FlexibleMink\PseudoInterface\LinkContextInterface;
+
+trait LinkContext
```
Comment (ramsey-medology) [inline src/Behat/FlexibleMink/Context/LinkContext.php]: Provide a description.

File: src/Behat/FlexibleMink/Context/LinkContext.php [hunk 5]
```diff
@@ -5,6 +5,9 @@
 use Behat\FlexibleMink\PseudoInterface\FlexibleContextInterface;
 use Behat\FlexibleMink\PseudoInterface\LinkContextInterface;
 
+/**
+ * {@inheritdoc}
+ */
```
Comment (Chekote) [inline src/Behat/FlexibleMink/Context/LinkContext.php]: What is this supposed to be inheriting? there's no extending or implementing happening here.

Comment (juanmrad) [reply src/Behat/FlexibleMink/Context/LinkContext.php]: All other classes had it like that 😆 added a better description.

File: features/assertLinkLocation.feature (hunk 2 above)
Comment (juanmrad) [reply features/assertLinkLocation.feature]: sorry missed xD.

**Pull request merged**